NDTensors/ext/NDTensorsCUDAExt/mul.jl

```diff
@@ -0,0 +1,27 @@
+# This was calling generic matrix multiplication.
+# TODO: Raise an issue with `CUDA.jl`.
+function mul!(
```

Comment:
```suggestion
function LinearAlgebra.mul!(
```
Let's use `LinearAlgebra.mul!` for overloading instead of `import LinearAlgebra: mul!`.

I also like that you started listed what we are using explicitly like:
```julia
using LinearAlgebra: LinearAlgebra, Adjoint, Transpose
```
so we should continue with that style of bringing types and functions into the namespace of these submodules/extensions, so then we would add `mul!` there to make use of it.

Reply:
I saw you use this in `NDTensorsMetal` and I think that it makes sense, I also like explicitly writing the package function we are overloading in the extensions, I think its nice natural documentation.  I can continue doing this in the future. I have made the modifications to the `mul!` functions in NDTensorsCUDA

Reply:
Sounds good, thanks. The importing style is mixed right now in `NDTensors` and `ITensors` because at some point I switched over entirely to using `import Base: *` but I think we should start to move away from that since it can be confusing when reading library code where a function is coming from, so maybe with new code we write we can start using `Base.*` for overloading.